detection_rules/rule.py

```diff
@@ -230,6 +230,16 @@ class AlertSuppressionDuration:
 
     group_by: List[definitions.NonEmptyStr]
     duration: Optional[AlertSuppressionDuration] = field(metadata=dict(metadata=dict(min_compat="8.7")))
+    missing_fields_strategy: Optional[definitions.AlertSuppressionMissing] = field(
+        metadata=dict(metadata=dict(min_compat="8.8")))
+
+    def validate(self, data: Any):
+        """Validate rule attributes with alert suppression."""
+
+        # validate language is 'kuery' if alert suppression is used
+        language = data.get('language')
```

Comment:
so, good news - alert suppression seems to only be compatible with the `query` rule type, rather than just the kql language. That simplifies how you would approach this.

Essentially, if you just move the `alert_supression` field from `BaseRuleData` into `QueryRuleData`, you can remove all the rest of the code, since it would enforce the valid field only for that rule type.

Reply:
Moving this into `QueryRuleData` with `eql` set as type and language, still load through the schema.